src/lib.rs

```diff
@@ -1817,27 +1817,57 @@ pub fn process_frame(sequence: &mut Sequence, fi: &mut FrameInvariants,
             let packet = encode_frame(sequence, fi, &mut fs);
             write_ivf_frame(output_file, fi.number, packet.as_ref());
             if let Some(mut y4m_enc) = y4m_enc {
-                let mut rec_y = vec![128 as u8; width * height];
-                let mut rec_u = vec![128 as u8; width * height / 4];
-                let mut rec_v = vec![128 as u8; width * height / 4];
-                for (y, line) in rec_y.chunks_mut(width).enumerate() {
-                    for (x, pixel) in line.iter_mut().enumerate() {
-                        let stride = fs.rec.planes[0].cfg.stride;
-                        *pixel = fs.rec.planes[0].data[y*stride+x] as u8;
+                let pitch_y = if sequence.bit_depth > 8 {
+                    width * 2
+                } else {
+                    width
+                };
+                let pitch_uv = pitch_y / 2;
+
+                let (mut rec_y, mut rec_u, mut rec_v) = (
+                    vec![128u8; pitch_y * height],
+                    vec![128u8; pitch_uv * (height / 2)],
+                    vec![128u8; pitch_uv * (height / 2)]);
+                
+                let (stride_y, stride_u, stride_v) = (
+                    fs.rec.planes[0].cfg.stride,
+                    fs.rec.planes[1].cfg.stride,
+                    fs.rec.planes[2].cfg.stride);
+
+                for (y, line) in fs.rec.planes[0].data.chunks_mut(stride_y).enumerate() {
```

Comment:
Rec does not mutate and you can use `chunks_mut()` on `rec_y` and friends.

``` rust
for (line, line_out) in fs.rec.planes[0].data.chunks().zip(rec_y.chunks_mut(pitch_y))
```

Reply:
Thank you.